src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -968,6 +968,7 @@ private Task ProcessGoAwayFrameAsync()
         }
 
         StopProcessingNextRequest(serverInitiated: false);
```

Comment:
```suggestion
        // StopProcessingNextRequest must be called before RequestClose to ensure it's considered client initiated.
        StopProcessingNextRequest(serverInitiated: false);
```

Let's add a comment to make it clear order matters here because of [the ConnectionClosedRequested registration in HttpConnection](https://github.com/dotnet/aspnetcore/blob/e971d938793f2d0b8bc96e5f839e61066d19ecea/src/Servers/Kestrel/Core/src/Internal/HttpConnection.cs#L101).